### ISSUE TITLE
EntityMetadataNotFoundError when using custom TypeORM model

### ISSUE THREAD

**Issue by fandy** [nextauthjs/adapters#228]
Trying to rename some columns in the default schema such that they all use snake case. I have the below schema, adjusted from the docs to remap the names to snake case:

```js
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  OneToMany,
  ValueTransformer,
} from "typeorm";

const transformer: Record<"date" | "bigint", ValueTransformer> = {
  date: {
    from: (date: string | null) => date && new Date(parseInt(date, 10)),
    to: (date?: Date) => date?.valueOf().toString(),
  },
  bigint: {
    from: (bigInt: string | null) => bigInt && parseInt(bigInt, 10),
    to: (bigInt?: number) => bigInt?.toString(),
  },
};

@Entity({ name: "user" })
export class UserEntity {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ type: "varchar", nullable: true })
  name!: string | null;

  @Column({ type: "varchar", nullable: true, unique: true })
  email!: string | null;

  @Column({ type: "varchar", nullable: true, transformer: transformer.date })
  emailVerified!: string | null;

  @Column({ type: "varchar", nullable: true })
  image!: string | null;

  @OneToMany(() => SessionEntity, (session) => session.userId)
  sessions!: SessionEntity[];

  @OneToMany(() => AccountEntity, (account) => account.userId)
  accounts!: AccountEntity[];
}

@Entity({ name: "account" })
export class AccountEntity {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ type: "uuid" })
  userId!: string;

  @Column({ type: "text" })
  type!: string;

  @Column({ type: "text" })
  provider!: string;

  @Column({ type: "text" })
  providerAccountId!: string;

  @Column({ type: "varchar", nullable: true })
  refresh_token!: string;

  @Column({ type: "varchar", nullable: true })
  access_token!: string | null;

  @Column({
    nullable: true,
    type: "bigint",
    transformer: transformer.bigint,
  })
  expires_at!: number | null;

  @Column({ type: "varchar", nullable: true })
  token_type!: string | null;

  @Column({ type: "varchar", nullable: true })
  scope!: string | null;

  @Column({ type: "varchar", nullable: true })
  id_token!: string | null;

  @Column({ type: "varchar", nullable: true })
  session_state!: string | null;

  @Column({ type: "varchar", nullable: true })
  oauth_token_secret!: string | null;

  @Column({ type: "varchar", nullable: true })
  oauth_token!: string | null;

  @ManyToOne(() => UserEntity, (user) => user.accounts, {
    createForeignKeyConstraints: true,
  })
  user!: UserEntity;
}

@Entity({ name: "session" })
export class SessionEntity {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ type: "text", unique: true })
  sessionToken!: string;

  @Column({ type: "uuid" })
  userId!: string;

  @Column({ transformer: transformer.date, type: "date" })
  expires!: string;

  @ManyToOne(() => UserEntity, (user) => user.sessions)
  user!: UserEntity;
}

@Entity({ name: "verification_token" })
export class VerificationTokenEntity {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ type: "text" })
  token!: string;

  @Column({ type: "text" })
  identifier!: string;

  @Column({ transformer: transformer.date, type: "date" })
  expires!: string;
}

```

However, this gives me `error - EntityMetadataNotFoundError: No metadata for "UserEntity" was found.`. Is there something wrong with the schema or is this a bug with the adapter?

nextauth.ts:
```js
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";
import EmailProvider from "next-auth/providers/email";
import { TypeORMLegacyAdapter } from "@next-auth/typeorm-legacy-adapter";
import * as entities from "lib/entities";

export default NextAuth({
  providers: [
    EmailProvider({
      server: process.env.EMAIL_SERVER,
      from: process.env.EMAIL_FROM,
    }),
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
    }),
  ],
  adapter: TypeORMLegacyAdapter(
    {
      url: process.env.DATABASE_URL,
      type: "postgres",
      ssl: {
        ca: process.env.DATABASE_CERT,
        rejectUnauthorized: process.env.DOPPLER_ENV === "dev" ? false : true,
      },
    },
    { entities }
  ),
  debug: true,
});

```

**Comment by balazsorban44** [comment]
Just a guess, but maybe changing the naming of the default fields might require a change of the actual adapter code as well, because the adapter methods aren't necessarily aware of your renaming of the entities. I'll have to check this theory though.

**Comment by jrasanen** [comment]
@balazsorban44 CamelCase is awkward to use with SQL.

This is because most of the SQL databases use case-insensitive identifiers by default (I believe it's in the SQL standard, but feel free to correct me) and the developer has to escape all columns using `"`

```sql
create table foo ("UserId" text);

insert into foo (UserId) values ('Hello'); --  ERROR: column "userid" of relation "foo" does not exist
insert into foo ("UserId") values ('Hello'); -- works as expected
```

vs

```sql
create table foo (user_id text);

insert into foo (user_id) values ('Hello'); -- works
insert into foo ("user_id") values ('Hello'); -- works
```

---

Regarding the issue itself: I also get the error, though the exception isn't always pointing to `UserEntity` for me.

For me it sometimes is:

1) `[next-auth][error][SESSION_ERROR]
https://next-auth.js.org/errors#session_error No metadata for "SessionEntity" was found. EntityMetadataNotFoundError: No metadata for "SessionEntity" was found.`
2) `[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR]
https://next-auth.js.org/errors#oauth_callback_handler_error No metadata for "AccountEntity" was found. EntityMetadataNotFoundError: No metadata for "AccountEntity" was found.`

I'm not sure, but could the issue be in how TypeORM registers the entities?

In our case, we initialize the TypeORM connection, and use this in the application, then NextAuth's adapter initializes a second TypeORM connection registering its entities.

**Comment by balazsorban44** [comment]
Fixed in `0.1.2-next.17`, as of @jrasanen suggestions through #257
